### PR TITLE
[feature] The compensation bot responds to '++salary'

### PR DESCRIPTION
**Pull request by andrewtdiz**
### Adding basics of the compensation bot

**Related Issue**: 

++salary now DMs the user with a prompt to enter salary information

Also, MessageEmbeds were not functioning without updating to DiscordJS version 12, which has also been done on this PR.

### Pull request checklist

- [x] Branch and PR are named correctly
- [x] Updated relevant documentation
- [x] Tested with a tester bot
- [ ] Wrote unit tests for changes

### Testing instructions

Run bot and type: 
`++salary` in a channel or in DMs
Then check DMs

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ module.exports = {
     'jest/no-disabled-tests': 'error',
     'jest/no-focused-tests': 'error',
     'jest/no-identical-title': 'warn',
-    'linebreak-style': ['error', 'unix'],
+    'linebreak-style': ['error', 'windows'],
```
Comment (jackwherry) [inline .eslintrc.js:18]: Why are we switching to CRLF? .editorconfig defines Unix line endings for this project

File: src/bots/comp/embed.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import Discord from 'discord.js';
+// const embed = new Discord.MessageEmbed()
```
Comment (jackwherry) [inline src/bots/comp/embed.js:2]: We should probably try to avoid commented-out code getting committed

File: src/bots/comp/embed.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    {
+      example: '$5,000',
+      name: 'Relocation/Signing Bonus',
+    },
+    {
+      example: '80k/year',
+      name: 'Total Compensation',
+    },
+  ],
+  compensationFormatter: function (fields, isExample) {
```
Comment (jackwherry) [inline src/bots/comp/embed.js:53]: We've been using ES6 arrow notation elsewhere

File: src/bots/comp/embed.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  .setColor('#0099ff')
+  .setTitle(compEmbed.title)
+  .setAuthor('CS Careers Hub Official Bot')
+  .setDescription(compEmbed.description())
+  .addFields(
+    { name: 'Example:', value: compEmbed.compensationFormatter(compEmbed.compensationFields, true) },
+  );
+
+
+export default embed;
```
Comment (jackwherry) [inline src/bots/comp/embed.js:94]: Missing newline

**Review by jackwherry — CHANGES_REQUESTED** [review]
Mostly looks good, couple of minor things

**Comment by jackwherry** [comment]
One more thing: the PR name should start with [feature] written out

File: .eslintrc.js (hunk 1 above)
Comment (andrewtdiz) [reply .eslintrc.js:18]: I was getting errors in es-lint unless I changed this. my bad.

File: src/bots/comp/embed.js (hunk 3 above)
Comment (andrewtdiz) [reply src/bots/comp/embed.js:53]: I believe there are issues with using 'this' with an arrow function, so I try to steer clear of that.

File: .eslintrc.js (hunk 1 above)
Comment (jackwherry) [reply .eslintrc.js:18]: No worries! It might be an issue in the lint setup if correctly-formatted files throw errors on Windows.

Comment (sirNikolai) [reply .eslintrc.js:18]: because this is running on ubuntu we need this to be kept as unix

File: package.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
     "prettier": "^2.2.1"
   },
   "dependencies": {
-    "discord.js": "^11.1.0",
+    "discord.js": "12.5.1",
```
Comment (sirNikolai) [inline package.json:30]: caret

File: src/bots/comp/embed.js (hunk 3 above)
Comment (jackwherry) [reply src/bots/comp/embed.js:53]: Oh yeah, it does look like it can get a little [complicated](https://www.codementor.io/@dariogarciamoya/understanding-this-in-javascript-with-arrow-functions-gcpjwfyuc) in certain cases. If this code doesn't work with arrow functions, then that's a plenty good reason not to use them :)

File: src/bots/comp/embed.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    return str;
+  },
+  descriptionFields: [
+    'Thank you for providing offer information from CSCH Chat Bot!',
+    'We appreciate users like you who offer to give back to the community.',
+    'All information is always **anonymous**. \n',
+    'Please provide salary information in the format below:',
+  ],
+  title: 'Salary Reporter',
+};
```
Comment (brendacs) [inline src/bots/comp/embed.js:81]: I would like to move all these strings in this object out to a constants file within salary bot. That way, you can also use ES6 string concatenation easily, which would be much cleaner.

File: src/bots/comp/embed.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+
+const embed = new Discord.MessageEmbed()
+  .setColor('#0099ff')
```
Comment (brendacs) [inline src/bots/comp/embed.js:85]: We should set a constant for colors too so our embeds stay consistent. If this will differ by bot, we can set it in each bot's constant file. Otherwise, we can set it in the core constants file.

Comment (andrewtdiz) [reply src/bots/comp/embed.js:85]: Ideally there should be an embed folder in /utils

File: src/bots/comp/embed.js (hunk 6 above)
Comment (jackwherry) [reply src/bots/comp/embed.js:81]: We should do strings the same way as you do in stop bot https://github.com/brendacs/stop-bot/blob/dc9bae3438ef6907b98eae5158d73588a021ffcd/src/constants.js#L35

File: src/bots/comp/embed.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  .setTitle(compEmbed.title)
+  .setAuthor('CS Careers Hub Official Bot')
+  .setDescription(compEmbed.description())
+  .addFields(
+    { name: 'Example:', value: compEmbed.compensationFormatter(compEmbed.compensationFields, true) },
+  );
```
Comment (brendacs) [inline src/bots/comp/embed.js:91]: Maybe out of the scope of this PR, but I would like to make discord embeds reusable and move it to the utils folder... we can break this out into a separate PR though

**Review by brendacs — CHANGES_REQUESTED** [review]
Good start! Needs some changes

File: src/bots/comp/embed.js (hunk 7 above)
Comment (brendacs) [reply src/bots/comp/embed.js:85]: Yup, exactly my thoughts

File: src/bots/comp/embed.js (hunk 6 above)
Comment (brendacs) [reply src/bots/comp/embed.js:81]: Ick, stop bot code is a mess but there are some good things I guess

File: src/client.js [hunk 9]
```diff
@@ -33,7 +33,7 @@ client
 
 // Login client
 if (ENV !== envs.TESTING) {
-  client.login(BOT_TOKEN);
+  client.login('Nzg1Mjk0NjIyMDc5MTg4OTkz.X81wmw.OC-D719p49XAegJs0h3yzwTZhtw');
```
Comment (jackwherry) [inline src/client.js:36]: Definitely don't commit secrets into the repo

File: src/bots/comp/constants.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    return str;
+  },
+  descriptionFields: [
+    'Thank you for providing offer information from CSCH Chat Bot!',
+    'We appreciate users like you who offer to give back to the community.',
+    'All information is always **anonymous**. \n',
+    'Please copy and paste what\'s **inside** the codeblock below:',
+  ],
+  title: 'Salary Reporter',
+};
```
Comment (brendacs) [inline src/bots/comp/constants.js:68]: again, move all these strings to a constants file within salary bot

File: src/bots/comp/constants.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+export default compConstants;
```
Comment (brendacs) [inline src/bots/comp/constants.js:70]: newline

File: src/bots/comp/constants.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    if(!isExample) str += '```';
+    str += '\n';
+    str += '++salary';
+    str += '\n';
+    fields.forEach(field => {
+      str += field.name + ': ';
+      if (isExample) str += field.example;
+      str += '\n';
+    });
+    if(!isExample) str += '```';
```
Comment (brendacs) [inline src/bots/comp/constants.js:48]: i think we should just get rid of these ticks in general. it'll look pretty enough without

File: src/bots/comp/constants.js (hunk 11 above)
Comment (andrewtdiz) [reply src/bots/comp/constants.js:70]: What's the fix here? Delete the new line or add another?

File: package.json (hunk 5 above)
Comment (brendacs) [reply package.json:31]: This still needs a caret

**Comment by brendacs** [comment]
Can you undelete .env.template?

File: .env.template [hunk 13]
```diff
@@ -1,3 +0,0 @@
-BOT_TOKEN=yourbottoken
```
Comment (sirNikolai) [inline .env.template:1]: Why was this all removed?

Comment (andrewtdiz) [reply .env.template:1]: That was another mistake, my bad. I need to get better with this tech stack. I'm stepping down responsibility until I have a few more solid commits under my belt.

File: package.json [hunk 14]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "discord.js": "^11.1.0",
```
Comment (brendacs) [inline package.json:33]: this one should be removed?

File: .env.template (hunk 13 above)
Comment (jackwherry) [reply .env.template]: It's not a big deal. I think you've got the environment variables and line endings worked out. Just make sure to have a very good idea of what's going into every commit by reading over the diffs.

**Review by brendacs — APPROVED** [review]
